### DISCUSSION TITLE
[Bug]: Battery save files aren't recognized, .sav files don't automatically load when "Local mode" and "Cable" are checked in the "Link" options

### DISCUSSION THREAD

**Discussion by ghost** [visualboyadvance-m/visualboyadvance-m#1229] [General]
### Expected behavior?

Usually when launching the VBA-M emulator and opening a GameBoy Advance ROM, the battery save (.sav) file for a given ROM is automatically loaded and the game begins normally.

![Screenshot 2024-03-01 204552](https://github.com/visualboyadvance-m/visualboyadvance-m/assets/161918938/4fc1ff26-796c-4a0f-8269-e251a0ddcf0b)
![Screenshot 2024-03-01 191509](https://github.com/visualboyadvance-m/visualboyadvance-m/assets/161918938/18a08f6f-2b1c-4e85-8195-30ffc1d086bf)
![Screenshot 2024-03-01 201540](https://github.com/visualboyadvance-m/visualboyadvance-m/assets/161918938/4936a7c2-0f26-4f42-899c-cf7432daa20a)
![Screenshot 2024-03-01 191607](https://github.com/visualboyadvance-m/visualboyadvance-m/assets/161918938/828eea34-f5f8-4506-a220-870aef7fcae6)
![Screenshot 2024-03-01 201608](https://github.com/visualboyadvance-m/visualboyadvance-m/assets/161918938/7ff81e5c-7a92-4a92-9ade-aa8778626fe6)
![Screenshot 2024-03-01 191627](https://github.com/visualboyadvance-m/visualboyadvance-m/assets/161918938/9ceeb7f8-d3a8-40b1-bc65-9330cdde0646)
![Screenshot 2024-03-01 191714](https://github.com/visualboyadvance-m/visualboyadvance-m/assets/161918938/3427f8bb-d7b1-4b54-a558-6c1030a3d42f)

### Actual behavior?

However, when I adjust the Link settings by going to Options in the VBA-M Menu and enabling the "Local mode" and "Cable" settings... and after I quit and re-launch the emulator... the battery save (.sav) file for the game isn't recognized and doesn't automatically load when I restart the game; the emulator treats it as if I were starting a new game entirely. 

This bug had been frustrating me for weeks before I found out what was causing it. With every game I played I had to manually import and export the battery files and it got really tedious and tiresome repeating it every single time.

![Screenshot 2024-03-01 201013](https://github.com/visualboyadvance-m/visualboyadvance-m/assets/161918938/6ff2e4fa-de85-44bc-a276-67fd681ebf8a)
![Screenshot 2024-03-01 201040](https://github.com/visualboyadvance-m/visualboyadvance-m/assets/161918938/c3fe7f6d-91ed-486b-95bd-a7978a39be9c)
![Screenshot 2024-03-01 201115](https://github.com/visualboyadvance-m/visualboyadvance-m/assets/161918938/ff497454-8d6b-449c-956d-c28697dcdbfc)
![Screenshot 2024-03-01 211338](https://github.com/visualboyadvance-m/visualboyadvance-m/assets/161918938/afe22c3d-6dc5-43d8-841f-d83b8c0664a5)
![Screenshot 2024-03-01 201219](https://github.com/visualboyadvance-m/visualboyadvance-m/assets/161918938/34f8908a-5ba2-43aa-bd4a-2b4cceed3d70)
![Screenshot 2024-03-01 201250](https://github.com/visualboyadvance-m/visualboyadvance-m/assets/161918938/ec989177-5a23-4cfc-94c0-3e48fab3c7ca)
![Screenshot 2024-03-01 201312](https://github.com/visualboyadvance-m/visualboyadvance-m/assets/161918938/21ba58cd-f757-4a8d-a356-83893d6c7bef)
![Screenshot 2024-03-01 201346](https://github.com/visualboyadvance-m/visualboyadvance-m/assets/161918938/51835d5b-ff13-4d99-a280-410137500d72)
![Screenshot 2024-03-01 201403](https://github.com/visualboyadvance-m/visualboyadvance-m/assets/161918938/4f008162-233f-4f57-a2f8-940f7a8f5a45)
![Screenshot 2024-03-01 202358](https://github.com/visualboyadvance-m/visualboyadvance-m/assets/161918938/76041b80-c5b5-41cb-843c-5cf2248a2fd5)

### Steps to reproduce the problem

1. Launch the VBA-M program
2. Open "Options" in the Menu
3. Go to "Link" in "Options"
4. Check "Local mode" in "Link"
5. Go to "Type" in "Link"
6. Check "Cable" in "Type"
7. Go to "File" in the Menu
8. Click "Quit" and close emulator
9. Re-launch emulator
10. Open "File" in the Menu
11. Click "Open" in "File"
12. Select and open GameBoy Advance ROM
13. Wait and see that the battery save doesn't load

### Error messages?

N/A

### What game were you playing

Title: Pokémon Emerald
GB, GBC or GBA: GBA
ROM CODE: BPEE
BIOS Used?: None
SAVE TYPE: Flash 128 K
Clean, Intro'd or Hacked?: Clean

### What settings are you using?

Video driver (simple, opengl): OpenGL
Sound driver (directsound, xaudio, openal, SDL): OpenAL
Scale Filter Used?: 2x

Link: Local mode
Link Type: Cable

### Specifications and Build details

VBA-M: v2.1.9 (64-bit)
OS: WIndows 11 Version 22H2 (22621.3155)
RAM: 8.00 GB

### Copyright Compliance

- [X] There are no infringing files attached to this bug report.

**Comment by ghost** [comment]
I should also add that after some testing, I learned that the same thing happens (that is, the emulator not recognizing and not automatically loading the .sav battery save file) when I set the "Type" in the "Link" option to "Game Boy"

![Screenshot 2024-03-01 222652](https://github.com/visualboyadvance-m/visualboyadvance-m/assets/161918938/2c0beeda-a707-4116-a5c8-daf987b43d3d)

**Comment by Squall-Leonhart** [comment]
This is behaviour inherited from vba-link, with some alterations.

The save filename changes structure when linking is active from Filename.sav to Filename-#.sav, right now you will have to copy, paste and alter the files manually due to known https://github.com/visualboyadvance-m/visualboyadvance-m/issues/867 issue for the batteries to persist.
